packages/powersync/bin/setup_web.dart

```diff
@@ -0,0 +1,129 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:pubspec_parse/pubspec_parse.dart';
+
+void main(List<String> arguments) async {
+  // Add a flag to enable/disable the download of worker
+  // Pass the no_worker argument to disable the download of the worker
+  // dart run powersync:setup_web no_worker
+  bool downloadWorker = true;
+  if (arguments.contains("no_worker")) {
+    downloadWorker = false;
+  }
+
+  final root = Directory.current.uri;
+  print('Project root: ${root.toFilePath()}');
+
+  final wasmPath = '${root.toFilePath()}web/sqlite3.wasm';
+
+  final workerPath = '${root.toFilePath()}web/powersync_db.worker.js';
+
+  final packageConfigFile = File.fromUri(
+    root.resolve('.dart_tool/package_config.json'),
+  );
+  dynamic packageConfig;
+  try {
+    packageConfig = json.decode(await packageConfigFile.readAsString());
+  } on FileSystemException {
+    print('Missing .dart_tool/package_config.json');
+    print('Run `flutter pub get` first.');
+    exit(1);
+  } on FormatException {
+    print('Invalid .dart_tool/package_config.json');
+    print('Run `flutter pub get` first.');
+    exit(1);
+  }
+
+  final powersyncPackageName = 'powersync';
+
+  final powersyncPkg =
+      getPackageFromConfig(packageConfig, powersyncPackageName);
+
+  final powersyncVersion =
+      getPubspecVersion(packageConfigFile, powersyncPkg, powersyncPackageName);
+
+  final sqlitePackageName = 'sqlite3';
+
+  final sqlite3Pkg = getPackageFromConfig(packageConfig, sqlitePackageName);
+
+  String sqlite3Version =
+      "v${getPubspecVersion(packageConfigFile, sqlite3Pkg, sqlitePackageName)}";
+  final httpClient = HttpClient();
+
+  String latestTag = await getLatestTagFromRelease(httpClient);
```

Comment:
The logic here doesn't seem complete. 

It looks like it's getting the latest release tag from all the releases and then using it if the tag starts with the `sqlite3Version`. 

This might be okay for some scenarios, but it will not give the latest `-powersync.x` version if the `sqlite3Version` in use in the application does not match our latest release. For example if the application is using `sqlite3:2.3.0` and our latest release is `2.4.0-powersync.1` but we also had `2.3.0-powersync.4` then it will only use `2.3.0-powersync.1`

Reply:
I reworked it to match a tag from the github releases, if nothing is matched it uses the latest version.